complete/src/main/java/dao/ITourGuideDAO.java

```diff
@@ -0,0 +1,14 @@
+
+
+public interface ITourGuideDAO {
+    TourGuide save(TourGuide tourGuide);
+
+    List<TourGuide> fetchAll();
+
+    @Nullable
+    TourGuide fetchById(int id);
+
+    void delete(int id);
+
+    List<TourGuide> fetchByGlobalSearch(String term);
+}
```

Comment:
Added Interface for DAO.

Source - https://www.tutorialspoint.com/java/java_data_structures.htm